### PR TITLE
patch vulnerabilities

### PR DESCRIPTION
**Pull request by mark-omarov**
Hi 👋 
The security department didn't like Snyk's report on this action, and I didn't like the idea of making a custom action for the same job, so here we go. We will be using the forked version for now, but you might also want to consider this.

Changes:
- Moved some dev dependencies from `dependencies` to `devDependencies`
- Forced resolution of vulnerable packages

I've tested these changes on my fork by [releasing the changes](https://github.com/mark-omarov/changesets-action/releases) and going through the [PR raised by the action](https://github.com/mark-omarov/changesets-action/pull/2). I might have missed something.

Cheers!

### CODE REVIEW

**Comment by MrTibbles** [comment]
This PR is also of great interest as we wish to use in our hosted Github enterprise runners but can not due to Snyk report. @Andarist are you able to provide any insights into how the approval process works, i am asking you as you made the most recent commit to the repo! :)

**Comment by mark-omarov** [comment]
There are conflicts needs to be resolved. I'm occupied with other things right now, but I'll try to get to it sometime this weekend.

Thanks everyone.

**Comment by mark-omarov** [comment]
I got a minute to resolve conflicts, did a scan with Dependabot, and added resolutions to patch reported vulnerabilities. As of now, there should be no vulnerabilities.

Sorry to bother @Andarist, but I'd appreciate it if you could take a look.
I tested the latest changes again on the fork, but don't quote me. I might have missed something.

**Comment by alex-ju** [comment]
hey @Andarist! sorry to keep nudging you on reviewing this, but I would hate for us (and potentially others) to have to fork away from this repo because of security compliance. it would be lovely if you could make some time to review this 🙏

**Comment by mark-omarov** [comment]
Hi @emmatown , sorry to bother you, in the hope of getting this PR noticed, I figured you might be able to help. If no - apologies for annoyance. ☺️

File: package.json [hunk 1]
```diff
@@ -54,6 +54,14 @@
...
-    "**/@octokit/core": "4.2.0"
+    "**/@octokit/core": "4.2.0",
+    "glob-parent": "^5.1.2",
+    "node-fetch": "^2.6.7",
+    "minimist": "^1.2.8",
+    "mixme": "^0.5.2",
+    "minimatch": "^3.1.1",
+    "lodash": "^4.17.21",
+    "trim": "^0.0.3",
+    "y18n": "^4.0.1"
```
Comment (Andarist) [inline package.json:65]: Do we have a breakdown of what packages depend on some outdated versions of those? Couldn't we update some packages to fix this instead of overriding what is installed using `resolutions`?

Comment (Andarist) [inline package.json:65]: This is a dependency of `yargs` and `smartwrap` depends on it. It's a bummer because we actually don't depend on any parts of `smartwrap` that depend on that... the problem is that `smartwrap` ships both its CLI and a runtime library in a single package.

Comment (Andarist) [reply package.json:65]: Hm, I also don't quite see that package in the Snyk report.

File: package.json [hunk 2]
```diff
@@ -54,6 +54,14 @@
...
   "resolutions": {
-    "**/@octokit/core": "4.2.0"
+    "**/@octokit/core": "4.2.0",
+    "glob-parent": "^5.1.2",
+    "node-fetch": "^2.6.7",
+    "minimist": "^1.2.8",
+    "mixme": "^0.5.2",
+    "minimatch": "^3.1.1",
+    "lodash": "^4.17.21",
+    "trim": "^0.0.3",
```
Comment (Andarist) [inline package.json:64]: This one comes from `remark-parse` and to fix it we'd have to update a major of this library. I agree that it's best to do it in a separate PR.

File: package.json [hunk 3]
```diff
@@ -54,6 +54,14 @@
   },
   "prettier": {},
   "resolutions": {
-    "**/@octokit/core": "4.2.0"
+    "**/@octokit/core": "4.2.0",
+    "glob-parent": "^5.1.2",
+    "node-fetch": "^2.6.7",
+    "minimist": "^1.2.8",
```
Comment (Andarist) [inline package.json:60]: I'm not sure what this one is about. I don't see any package to depend on this.

File: package.json [hunk 4]
```diff
@@ -54,6 +54,14 @@
...
+    "mixme": "^0.5.2",
+    "minimatch": "^3.1.1",
+    "lodash": "^4.17.21",
```
Comment (Andarist) [inline package.json:63]: after your updates packages don't depend on this at all so this is redundant (likely the same story for `minimist`)

File: .changeset/wicked-apes-watch.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": patch
+---
+
+Updated a few dependencies to patch the reported security vulnerabilities that were reported for their older versions.
```
Comment (Andarist) [inline .changeset/wicked-apes-watch.md:5]:
```suggestion
Updated a few dependencies to patch the security vulnerabilities that were reported for their older versions.
```

**Comment by Andarist** [comment]
I tweaked this PR a little bit because I think that some bits were overeager and that some of those "vulnerabilities" went away when some of the packages were updated.

Let's publish this and then recheck with Snyk again.

**Comment by mark-omarov** [comment]
Thanks, @Andarist !
I might have forgotten to rescan after resolving conflicts, some vulnerabilities might have indeed been resolved by then.
I'm gonna ask the security team to rescan next week.
